### PR TITLE
Add diagrams at Makefile

### PR DESCRIPTION
**Pull request by fao89**
[noissue]

### CODE REVIEW

File: templates/docs/docs/Makefile.j2 [hunk 1]
```diff
@@ -2,11 +2,13 @@
...
-PAPER         =
-BUILDDIR      = _build
-PULP_URL      = "http://localhost:24817"
+SPHINXOPTS    		  = -W -n
+SPHINXBUILD   		  = sphinx-build
+PAPER         		  =
+BUILDDIR      		  = _build
+DIAGRAM_BUILD_DIR     = _diagrams
+DIAGRAM_SOURCE_DIR    = diagrams_src
+PULP_URL      		  = "http://localhost:24817"
```
Comment (mdellweg) [inline templates/docs/docs/Makefile.j2:11]: Somehow, the alignment of the equals went wrong here? Is this some uncommon tab expansion?
I would not even bother to align them, as in "Why do i need to touch all the lines if one variable name becomes longer?"

File: templates/docs/docs/Makefile.j2 [hunk 2]
```diff
@@ -72,7 +85,7 @@ htmlhelp:
 	$(SPHINXBUILD) -b htmlhelp $(ALLSPHINXOPTS) $(BUILDDIR)/htmlhelp
 	@echo
 	@echo "Build finished; now you can run HTML Help Workshop with the" \
-	      ".hhp project file in $(BUILDDIR)/htmlhelp."
+		  ".hhp project file in $(BUILDDIR)/htmlhelp."
```
Comment (mdellweg) [inline templates/docs/docs/Makefile.j2:88]: Also some formatting strangeness around tabs?

**Review by mdellweg — CHANGES_REQUESTED** [review]
Would you mind explaining the reason for this change, as it goes as [noissue] (and is not obvious to me)?

**Comment by fao89** [comment]
> Would you mind explaining the reason for this change, as it goes as [noissue] (and is not obvious to me)?

I forgot to add it here, it was a fix for pulp_file,
PR: https://github.com/pulp/pulp_file/pull/431
issue: https://travis-ci.com/github/pulp/pulp_file/jobs/387174826#L1220

File: templates/docs/docs/Makefile.j2 (hunk 1 above)
Comment (fao89) [reply templates/docs/docs/Makefile.j2:11]: this was looking different on my editor, I think it was a space/tab issue with my editor config

File: templates/docs/docs/Makefile.j2 (hunk 2 above)
Comment (fao89) [reply templates/docs/docs/Makefile.j2:88]: his was looking different on my editor, I think it was a space/tab issue with my editor config

File: templates/docs/docs/Makefile.j2 (hunk 1 above)
Comment (fao89) [reply templates/docs/docs/Makefile.j2:11]: I checked it at vscode, sublime, and vim. And I couldn't reproduce it locally
image

**Review by mdellweg — APPROVED** [review]

**Review by dkliban — APPROVED** [review]

**Pull request merged**